internal/service/fw/api_access_codes_resource_test.go

```diff
@@ -0,0 +1,408 @@
+package fw_test
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/terraform"
+
+	"github.com/infobloxopen/bloxone-go-client/fw"
+	"github.com/infobloxopen/terraform-provider-bloxone/internal/acctest"
+)
+
+//TODO: add tests
+// The following require additional resource/data source objects to be supported.
+// - policy_ids
+
+func TestAccAccessCodesResource_basic(t *testing.T) {
+	var resourceName = "bloxone_td_access_code.test"
+	var v fw.AtcfwAccessCode
+	name := acctest.RandomNameWithPrefix("ac")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(t) },
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// Create and Read
+			{
+				Config: testAccAccessCodesBasicConfig(name),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAccessCodesExists(context.Background(), resourceName, &v),
+				),
+			},
+			// Delete testing automatically occurs in TestCase
+		},
+	})
+}
+
+func TestAccAccessCodesResource_disappears(t *testing.T) {
+	resourceName := "bloxone_td_access_code.test"
+	var v fw.AtcfwAccessCode
+	name := acctest.RandomNameWithPrefix("ac")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(t) },
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		CheckDestroy:             testAccCheckAccessCodesDestroy(context.Background(), &v),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAccessCodesBasicConfig(name),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAccessCodesExists(context.Background(), resourceName, &v),
+					testAccCheckAccessCodesDisappears(context.Background(), &v),
+				),
+				ExpectNonEmptyPlan: true,
+			},
+		},
+	})
+}
+
+func TestAccAccessCodesResource_Name(t *testing.T) {
+	resourceName := "bloxone_td_access_code.test_name"
+	var v1, v2 fw.AtcfwAccessCode
+	name1 := acctest.RandomNameWithPrefix("ac")
+	name2 := acctest.RandomNameWithPrefix("ac")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(t) },
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// Create and Read
+			{
+				Config: testAccAccessCodesName(name1),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAccessCodesExists(context.Background(), resourceName, &v1),
+					resource.TestCheckResourceAttr(resourceName, "name", name1),
+					resource.TestCheckResourceAttr(resourceName, "rules.0.data", "terraform_test"),
```

Comment:
`data` and `type` don't have to be verified here, makes it redundant. They should be in basic, and in the test that is specific to that field.

Reply:
Removed . Similar fields have been removed from other object too